### ISSUE TITLE
Only include enum34 if necessary (Python 3.6)

### ISSUE THREAD

**Issue by bradmwalker** [grpc/grpc#9900]
The enum34 package is [incompatible with Python 3.6's enum implementation](https://bitbucket.org/stoneleaf/enum34/issues/19/enum34-isnt-compatible-with-python-36). This breaks Python processes depending on import order. Uninstalling enum34 after installing grpcio fixes things.

Test case:
~~~
[brad@f25 ~]$ python3.6 -m venv grpc-enum-test
[brad@f25 ~]$ source grpc-enum-test/bin/activate
(grpc-enum-test) [brad@f25 ~]$ pip install grpcio pytest-xdist
Collecting grpcio
  Using cached grpcio-1.1.3-cp36-cp36m-manylinux1_x86_64.whl
Collecting pytest-xdist
Collecting protobuf>=3.0.0 (from grpcio)
  Using cached protobuf-3.2.0-cp36-cp36m-manylinux1_x86_64.whl
Collecting six>=1.5.2 (from grpcio)
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting enum34>=1.0.4 (from grpcio)
  Using cached enum34-1.1.6-py3-none-any.whl
Collecting execnet>=1.1 (from pytest-xdist)
  Using cached execnet-1.4.1-py2.py3-none-any.whl
Collecting pytest>=2.4.2 (from pytest-xdist)
  Using cached pytest-3.0.6-py2.py3-none-any.whl
Collecting py>=1.4.22 (from pytest-xdist)
  Using cached py-1.4.32-py2.py3-none-any.whl
Requirement already satisfied: setuptools in ./grpc-nemu-test/lib/python3.6/site-packages (from protobuf>=3.0.0->grpcio)
Collecting apipkg>=1.4 (from execnet>=1.1->pytest-xdist)
  Using cached apipkg-1.4-py2.py3-none-any.whl
Installing collected packages: six, protobuf, enum34, grpcio, apipkg, execnet, py, pytest, pytest-xdist
Successfully installed apipkg-1.4 enum34-1.1.6 execnet-1.4.1 grpcio-1.1.3 protobuf-3.2.0 py-1.4.32 pytest-3.0.6 pytest-xdist-1.15.0 six-1.10.0
(grpc-enum-test) [brad@f25 ~]$ cd grpc-enum-test 
(grpc-enum-test) [brad@f25 grpc-enum-test]$ py.test -f
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 3, in <module>
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/execnet/gateway_base.py", line 18, in <module>
    import traceback
  File "/usr/lib64/python3.6/traceback.py", line 5, in <module>
    import linecache
  File "/usr/lib64/python3.6/linecache.py", line 11, in <module>
    import tokenize
  File "/usr/lib64/python3.6/tokenize.py", line 33, in <module>
    import re
  File "/usr/lib64/python3.6/re.py", line 142, in <module>
    class RegexFlag(enum.IntFlag):
AttributeError: module 'enum' has no attribute 'IntFlag'
Traceback (most recent call last):
  File "/home/brad/grpc-enum-test/bin/py.test", line 11, in <module>
    sys.exit(main())
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/_pytest/config.py", line 57, in main
    return config.hook.pytest_cmdline_main(config=config)
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/_pytest/vendored_packages/pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/_pytest/vendored_packages/pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/_pytest/vendored_packages/pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/_pytest/vendored_packages/pluggy.py", line 614, in execute
    res = hook_impl.function(*args)
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/xdist/looponfail.py", line 32, in pytest_cmdline_main
    looponfail_main(config)
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/xdist/looponfail.py", line 42, in looponfail_main
    remotecontrol.loop_once()
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/xdist/looponfail.py", line 111, in loop_once
    self.setup()
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/xdist/looponfail.py", line 73, in setup
    self.gateway = self.initgateway()
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/xdist/looponfail.py", line 65, in initgateway
    return execnet.makegateway("popen")
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/execnet/multi.py", line 128, in makegateway
    gw = gateway_bootstrap.bootstrap(io, spec)
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/execnet/gateway_bootstrap.py", line 92, in bootstrap
    bootstrap_import(io, spec)
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/execnet/gateway_bootstrap.py", line 27, in bootstrap_import
    s = io.read(1)
  File "/home/brad/grpc-enum-test/lib64/python3.6/site-packages/execnet/gateway_base.py", line 389, in read
    "expected %d bytes, got %d" % (numbytes, len(buf)))
EOFError: expected 1 bytes, got 0
~~~

**Comment by bradmwalker** [comment]
Actually uninstalling enum34 breaks starting a gRPC server. pkg_resources errors about needing enum34.

**Comment by revmischa** [comment]
The version currently on PyPI doesn't have this fix, consequently doesn't run on python 3.6. Please release a new version?
